src/client/Microsoft.Identity.Client/Platforms/iOS/iOSBroker.cs

```diff
@@ -180,9 +192,33 @@ private MsalTokenResponse ResultFromBrokerResponse(Dictionary<string, string> re
                 };
             }
 
+            if (!string.IsNullOrEmpty(BrokerRequestNonce))
```

Comment:
verification should be at the top of this method (if this fails, there's no point in doing all the other calculations). It should also encapsulate all the validatin logic, i.e. including line 195.